src/main/java/org/tugraz/sysds/runtime/instructions/fed/InitFEDInstruction.java

```diff
@@ -34,106 +35,126 @@
 
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.net.UnknownHostException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
 import java.util.concurrent.Future;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 public class InitFEDInstruction extends FEDInstruction {
 	private CPOperand _addresses, _ranges, _output;
-	
+
 	public InitFEDInstruction(CPOperand addresses, CPOperand ranges, CPOperand out, String opcode, String instr) {
 		super(FEDType.Init, opcode, instr);
 		_addresses = addresses;
 		_ranges = ranges;
 		_output = out;
 	}
-	
+
 	public static InitFEDInstruction parseInstruction(String str) {
 		String[] parts = InstructionUtils.getInstructionPartsWithValueType(str);
-		// We need 3 parts: Opcode, Addresses (list of Strings with url/ip:port/filepath), ranges and the output Operand
-		if( parts.length != 4 )
+		// We need 4 parts: Opcode, Addresses (list of Strings with
+		// url/ip:port/filepath), ranges and the output Operand
+		if (parts.length != 4)
```

Comment:
* this said already != 4 before ?!
* why not make it 5 and have the port in its own field?

Reply:
Because changing it at a different place, would make it parse the URL there, moving the functionality to another location in code.
The change  in the If line, is just formatting, putting a space after the if and before the (